tests/test_functional_api.py

```diff
@@ -0,0 +1,1251 @@
+
+    def test_api_history_delete_failed(self):
+        # Check the number of failed entries currently in the history
+        json = self._get_api_history({"failed_only": 1})
+        failed_count = len(json["history"]["slots"])
+        original_history_size = int(self.history_size)
+
+        if failed_count > 0:
+            # Remove all failed entries
+            json = self._get_api_history({"name": "delete", "value": "failed"})
+            assert json["status"] is True
+
+            # Verify no failed history entries remain
+            json = self._get_api_history()
+            for slot in json["history"]["slots"]:
+                assert slot["status"] != Status.FAILED
+
+            # Make sure nothing else got axed
+            new_history_size = original_history_size - failed_count
+            assert len(json["history"]["slots"]) == new_history_size
+        else:
+            warn("Fake history doesn't contain any failed jobs")
+            new_history_size = original_history_size
+
+        # A rerun of the delete action shouldn't have any effect, since no failed entries should remain
+        json = self._get_api_history({"name": "delete", "value": "failed"})
+        assert json["status"] is True
+        json = self._get_api_history()
+        assert len(json["history"]["slots"]) == new_history_size
+
+    def test_api_history_delete_completed(self):
+        json = self._get_api_history({"name": "delete", "value": "completed"})
+        assert json["status"] is True
+        json = self._get_api_history()
+        for slot in json["history"]["slots"]:
+            assert slot["status"] != Status.COMPLETED
+
+
+@pytest.mark.usefixtures("api_server", "update_history_specs")
+class TestHistoryApiPart2(ApiTestFunctions):
+    """Test history-related API responses, part 2. A separate testcase is
+    needed because the previous one ran out of history entries to delete."""
+
+    api_server_parameters = {
+        "add_fake_history": True,
+        "history_size": randint(42, 81),
+        "add_locale": False,
+    }
+
+    def test_api_history_delete_all(self):
+        json = self._get_api_history({"name": "delete", "value": "all"})
+        assert json["status"] is True
+        json = self._get_api_history()
+        for slot in json["history"]["slots"]:
+            assert slot["status"] != Status.COMPLETED
+            assert slot["status"] != Status.FAILED
+
+    def test_api_history_delete_everything(self):
+        json = self._get_api_history()
+        delete_these = [slot["nzo_id"] for slot in json["history"]["slots"]]
+        assert len(delete_these) == len(json["history"]["slots"])
+
+        # Kill 'm with fire!
+        json = self._get_api_history({"name": "delete", "value": ",".join(delete_these)})
+        assert json["status"] is True
+
+        # Make sure nothing survived
+        json = self._get_api_history()
+        assert json["history"]["noofslots"] is int(0)
```

Comment:
`== 0`?

Reply:
Done, also removed all other occurences of the same.